src/themes/default/pages/Product.vue

```diff
@@ -120,12 +128,25 @@ export default {
       }
     },
     addToCompare () {
-      // todo
-      this.$bus.$emit('notification', {
-        type: 'success',
-        message: 'Product has been added to comparison list. However - this feature is not implemented yet :(',
-        action1: { label: 'OK', action: 'close' }
-      })
+      let self = this
+      if (!self.compare.isCompare) {
+        this.$store.dispatch('compare/addItem', self.product).then(res => {
+          self.compare.isCompare = true
+        })
+      } else {
+        this.$store.dispatch('compare/removeItem', self.product).then(res => {
+          self.compare.isCompare = false
+        })
+      }
+    }
+  },
+  mounted: function () {
+    if (this.wishlistCheck.isOnWishlist(this.product)) {
+      this.favorite.icon = 'favorite'
+      this.favorite.isFavorite = true
+    }
+    if (this.compareCheck.isOnCompare(this.product)) {
+      this.compare.isCompare = true
```

Comment:
The problem you mentioned in the email regards the component lifecycle indeed. When you’re on product page and use Search or Related widgets, you stays on Product page but route is just changed

This is the reason “mounted” is not called.

Please move this logic to core/pages/Product.vue:loadData

As this method is static You not have access to “this” so what
You should done is to use ui-store vuex store keep the state of the buttons

This (method is static) is because it supports SSR and the asyncData method used for fetching the data server side, must be static